src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs

```diff
@@ -0,0 +1,78 @@
+using Microsoft.ApplicationInsights.Extensibility;
+using System.Management.Automation;
+using System.Security.Cryptography;
+using System.Collections.Generic;
+using System.Reflection;
+using System.IO;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// send up telemetry for startup
+    /// </summary>
+    public class ApplicationInsightsTelemetry
+    {
+        // The semaphore file which indicates whether telemetry should be sent
+        // This is temporary code waiting on the acceptance and implementation of the configuration spec
+        /// <summary>
+        /// The name of the file by when present in $PSHOME will enable telemetry.
+        /// If this file is not present, no telemetry will be sent.
+        /// </summary>
+        public const string TelemetrySemaphoreFilename = "DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY";
+
+        /// <summary>
+        /// The path to the semaphore file which enables telemetry
+        /// </summary>
+        public static string TelemetrySemaphoreFilePath = Path.Combine(
+            Utils.GetApplicationBase(Utils.DefaultPowerShellShellID),
+            TelemetrySemaphoreFilename);
+
+        // Telemetry client to be reused when we start sending more telemetry
+        private static TelemetryClient _telemetryClient = null;
+
+        // Set this to true to reduce the latency of sending the telemetry
+        private static bool _developerMode = false;
+
+        // PSCoreInsight2 telemetry key
+        private const string _psCoreTelemetryKey = "ee4b2115-d347-47b0-adb6-b19c2c763808";
+
+        /// <summary>
+        /// Send the telemetry
+        /// </summary>
+        private static void SendTelemetry(string eventName, Dictionary<string,string>payload)
+        {
+            try
+            {
+                // if the semaphore file exists, try to send telemetry
+                if (Utils.NativeFileExists(TelemetrySemaphoreFilePath))
+                {
+                    TelemetryConfiguration.Active.InstrumentationKey = _psCoreTelemetryKey;
+                    TelemetryConfiguration.Active.TelemetryChannel.DeveloperMode = _developerMode;
+                    if ( _telemetryClient == null )
+                    {
+                        _telemetryClient = new TelemetryClient();
+                    }
+                    _telemetryClient.TrackEvent(eventName, payload, null);
+                }
+            }
+            catch (Exception)
+            {
+                ; // Do nothing, telemetry can't be sent
+            }
+        }
+
+        /// <summary>
+        /// Create the startup payload and send it up
+        /// </summary>
+        public static void SendPSCoreStartupTelemetry()
+        {
+            var properties = new Dictionary<string, string>();
+            properties.Add("GitCommitID", PSVersionInfo.GitCommitId);
+            properties.Add("OSVersionInfo", Environment.OSVersion.VersionString);
```

Comment:
`Environment.OSVersion.VersionString` doesn't give much useful information on different distro of Linux, for example, it show `Unix 4.8.0.41` on Ubuntu 16.04. How about use `[System.Runtime.InteropServices.RuntimeInformation]::OSDescription`?
```
PS /> [System.Runtime.InteropServices.RuntimeInformation]::OSDescription                            
Linux 4.8.0-41-generic #44~16.04.1-Ubuntu SMP Fri Mar 3 17:11:16 UTC 2017
```